ballerina-tests/build.gradle

```diff
@@ -0,0 +1,186 @@
+    }
+    if (project.hasProperty('balNativeTest')) {
+        nativeFlag = '--native'
+    }
+
+    gradle.taskGraph.whenReady { graph ->
+        if (graph.hasTask(":${packageName}-ballerina:test")) {
+            if (!project.hasProperty('balNativeTest')) {
+                testParams = "${testCoverageParam}"
+            }
+        } else {
+            testParams = "--skip-tests"
+        }
+    }
+}
+
+task ballerinaTest {
+    dependsOn(":${packageName}-ballerina:build")
+    dependsOn(updateTomlFiles)
+    dependsOn(initializeVariables)
+    finalizedBy(commitTomlFiles)
+
+    doLast {
+        exec {
+            workingDir project.projectDir
+            environment "JAVA_OPTS", "-DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true"
+            if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                commandLine 'cmd', '/c', "${balJavaDebugParam} ${distributionBinPath}/bal.bat test ${nativeFlag} ${testParams} --offline ${groupParams} ${disableGroups} ${debugParams} && exit %%ERRORLEVEL%%"
+            } else {
+                commandLine 'sh', '-c', "${balJavaDebugParam} ${distributionBinPath}/bal test ${nativeFlag} ${testParams} --offline ${groupParams} ${disableGroups} ${debugParams}"
+            }
+        }
+    }
+}
+
+test {
+    dependsOn(ballerinaTest)
+}
+
+clean {
+    delete("${projectDir}/target")
+}
+
+
+task startNatsServers() {
+    doLast {
+        if (!Os.isFamily(Os.FAMILY_WINDOWS)) {
+            def stdOut = new ByteArrayOutputStream()
+            exec {
+                commandLine 'sh', '-c', "docker ps --filter name=server_nats-tls_1"
+                standardOutput = stdOut
+            }
+            if (!stdOut.toString().contains("server_nats-tls_1")) {
+                println "Starting NATS Basic server."
+                exec {
+                    commandLine 'sh', '-c', "docker-compose -f tests/server/docker-compose.yaml up -d"
+                    standardOutput = stdOut
+                }
+                println stdOut.toString()
+                sleep(5 * 1000)
+            } else {
+                println "NATS Basic server is already started."
+            }
+        }
+    }
+}
+
+task stopNatsServers() {
+    doLast {
+        if (!Os.isFamily(Os.FAMILY_WINDOWS)) {
+            def stdOut = new ByteArrayOutputStream()
+            exec {
+                commandLine 'sh', '-c', "docker ps --filter name=server_nats-tls_1"
+                standardOutput = stdOut
+            }
+            if (stdOut.toString().contains("server_nats-tls_1")) {
+                println "Stopping NATS server."
+                exec {
+                    commandLine 'sh', '-c', "docker-compose -f tests/server/docker-compose.yaml rm -svf"
+                    standardOutput = stdOut
+                }
+                println stdOut.toString()
+                sleep(5 * 1000)
+            } else {
+                println "NATS server is not started."
+            }
+        }
+    }
+}
+
+build.dependsOn startNatsServers
```

Comment:
We can optimize this like this: https://github.com/ballerina-platform/module-ballerina-oauth2/blob/ae45f422c7cb37f18699d75787d6124455ab93b6/ballerina/build.gradle#L217

This way, we don't need to start NATS servers when we don't need to run the tests.